src/libs/Navigation/Navigation.ts

```diff
@@ -187,32 +183,33 @@ function setParams(params, routeKey) {
 /**
  * Dismisses the last modal stack if there is any
  *
- * @param {String | undefined} targetReportID - The reportID to navigate to after dismissing the modal
+ * @param targetReportID - The reportID to navigate to after dismissing the modal
  */
-function dismissModal(targetReportID) {
+function dismissModal(targetReportID?: string) {
     if (!canNavigate('dismissModal')) {
         return;
     }
     const rootState = navigationRef.getRootState();
-    const lastRoute = _.last(rootState.routes);
-    switch (lastRoute.name) {
+    const lastRoute = rootState.routes.at(-1);
+    switch (lastRoute?.name) {
         case NAVIGATORS.RIGHT_MODAL_NAVIGATOR:
         case SCREENS.NOT_FOUND:
         case SCREENS.REPORT_ATTACHMENTS:
             // if we are not in the target report, we need to navigate to it after dismissing the modal
             if (targetReportID && targetReportID !== getTopmostReportId(rootState)) {
                 const state = getStateFromPath(ROUTES.REPORT_WITH_ID.getRoute(targetReportID));
 
-                const action = getActionFromState(state, linkingConfig.config);
-                action.type = 'REPLACE';
-                navigationRef.current.dispatch(action);
+                const action: CommonNavigationAction | StackActionType | undefined = getActionFromState(state, linkingConfig.config);
+                if (action) {
```

Comment:
Why is the logic here changed from the previous one? Or am I missing something?

Reply:
I think one line is missing indeed :)
```suggestion
                if (action) {
                action.type = 'REPLACE';
```

Reply:
Thanks, reverted 😄